src/App.tsx

```diff
@@ -15,37 +18,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  None,
+  Alphabetical,
+  Length,
+}
+
+function sortGoods(
+  input: string[],
+  sortType: SortType,
+  reverse: boolean,
+): string[] {
+  const preparedGoods = [...input];
+
+  if (sortType !== SortType.None) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.Alphabetical:
+          return good1.localeCompare(good2);
+        case SortType.Length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState(SortType.None);
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = sortGoods(goodsFromServer, sortType, reverse);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortType !== SortType.Alphabetical,
+          })}
+          onClick={() => setSortType(SortType.Alphabetical)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortType !== SortType.Length,
+          })}
+          onClick={() => setSortType(SortType.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reverse,
+          })}
+          onClick={() => {
+            setReverse(!reverse ? true : false);
```

Comment:
```suggestion
            setReverse(currentState => !currentState);
```
- It's already boolean, you don't need a ternary operator in this case
- When you set state depending on the previous one use functional `setState`